src/gurt/fault_inject.c

```diff
@@ -526,6 +526,8 @@ d_fault_inject_init(void)
 	pthread_once(&d_fi_gdata_init_once, d_fi_gdata_init);
 	D_ASSERT(d_fi_gdata.dfg_inited == 1);
 
+	/* Enable fault inject by default for dev mode */
+	d_fault_inject = 1;
```

Comment:
Should we add some conditional compile to distinguish whether it is under dev mode or release mode? such as:

```
#ifdef XXX_DEV
d_fault_inject = 1;
#else
d_fault_inject = 0;
```

Reply:
BUILD_TYPE=dev or release will use different d_fault_inject_xxxx  see src/gurt/fault_inject.c

Reply:
I mean:

```
#if FAULT_INJECTION
d_fault_inject = 1;
#else
d_fault_inject = 0;
#endif
```

Reply:
Just noticed that the whole d_fault_inject_init() has already been wrapped inside "#if FAULT_INJECTION".